src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private float threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        if (size > threshold) {
+            resize();
+        }
+        Node<K, V> oldNode = table[index];
+        while (oldNode.next != null
+                && !Objects.equals(key, oldNode.key)) {
+            oldNode = oldNode.next;
+        }
+        if (Objects.equals(key, oldNode.key)) {
+            oldNode.value = value;
+            return;
+        }
+        oldNode.next = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = table[getIndex(key)];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        int newLength = table.length * 2;
+        threshold = newLength * DEFAULT_LOAD_FACTOR;
+        Node<K, V>[] oldTable = table;
+        table = new Node[newLength];
+        size = 0;
+        for (Node<K, V> node : oldTable) {
```

Comment:
extract method transfer(Node<K, V>[] oldTable) for this code: `for (Node<K, V> node : oldTable) {
             while (node != null) {
                 put(node.key, node.value);
                 node = node.next;
             }
         }`